game.py

```diff
@@ -64,19 +66,29 @@ def put_down_card_on_table(self, player, card):
 
     def try_playing_matching_number_or_color_card(self, player):
         for card in player.cards:
-            if card.color == self.card_on_table.color or card.num == self.card_on_table.num:
+            if (card.color == self.card_on_table.color or card.num == self.card_on_table.num) and \
+                    not card.is_stop_card and not card.is_change_color:
```

Comment:
Another advantage of using an enum for the type of card, in this example, now, for each new type that you would like to add, you need to update this line even though it doesn't affect the new type. 

As general comment, white lists are usually better than a black list (what it affects on instead what it doesn't affect on)